tools/approval-service/internal/service/releaseservice.go

```diff
@@ -0,0 +1,294 @@
+/*
+ *  Copyright 2025 Gravitational, Inc
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package service
+
+import (
+	"cmp"
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/gravitational/shared-workflows/tools/approval-service/internal/config"
+	"github.com/gravitational/shared-workflows/tools/approval-service/internal/eventsources/githubevents"
+	teleportclient "github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/types"
+)
+
+// ReleaseService processes GitHub Events and Teleport Access Request state changes that are related to GitHub workflows.
+// At a high level, it waits for a GitHub Workflow to request access to an environment (signaled by deployment_protection_rule),
+// and then creates a Teleport Access Request for the workflow.
+// It then waits for the Access Request to be reviewed, and based on the review, it approves or rejects the workflow in GitHub.
+type ReleaseService struct {
```

Comment:
It does not appear that the release service business logic has sufficient test coverage. Can you add it?